onnxruntime/test/providers/qnn/einsum_op_test.cc

```diff
@@ -0,0 +1,190 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#if !defined(ORT_MINIMAL_BUILD)
+
+#include <string>
+#include <vector>
+
+#include "test/providers/qnn/qnn_test_utils.h"
+#include "core/graph/node_attr_utils.h"
+#include "test/util/include/test_utils.h"
+
+#include "core/graph/onnx_protobuf.h"
+#include "gtest/gtest.h"
+
+namespace {
+
+using onnxruntime::ProviderOptions;
+using onnxruntime::test::BuildOpTestCase;
+using onnxruntime::test::ExpectedEPNodeAssignment;
+using onnxruntime::test::RunQnnModelTest;
+using onnxruntime::test::TestInputDef;
+using onnxruntime::utils::MakeAttribute;
+
+template <typename DataType>
+static void RunQnnEinsum(
+    const std::string& backend,
+    const TestInputDef<DataType>& in0,
+    const TestInputDef<DataType>& in1,
+    const std::string& equation,
+    const float f32_abs_err = 1e-4f) {
+  ProviderOptions provider_options;
+  provider_options["backend_type"] = backend;
+  provider_options["offload_graph_io_quantization"] = "0";
+  RunQnnModelTest(
+      /*build_test_case=*/BuildOpTestCase<DataType, DataType>(
+          /*op_type=*/"Einsum",
+          /*input_defs_1=*/{in0, in1},
+          /*input_defs_2=*/{},
+          /*attrs=*/{MakeAttribute("equation", equation)}),
+      /*provider_options=*/provider_options,
+      /*opset_version=*/13,
```

Comment:
>13 [](http://example.com/codeflow?start=24&length=2)

Einsum is available since opset 12 and never changed, so set it to 12 here makes more sense.